executor/executor_issue_test.go

```diff
@@ -684,6 +684,11 @@ func TestIssue22231(t *testing.T) {
 	tk.MustQuery("select cast('2020-05-28 23:59:59 00:00:00' as datetime)").Check(testkit.Rows("2020-05-28 23:59:59"))
 	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1292 Truncated incorrect datetime value: '2020-05-28 23:59:59 00:00:00'"))
 	tk.MustExec("drop table if exists t_issue_22231")
+
+	tk.MustQuery("SELECT CAST(\"1111111111-\" AS DATE);")
+	tk.MustQuery("SHOW WARNINGS").
+		Check(
```

Comment:
Maybe combine L689-L691 to one line is better.

Reply:
Done